### DISCUSSION TITLE
Is there a connection manager you recommend?

### DISCUSSION THREAD

**Discussion by shi-yan** [rusqlite/rusqlite#1328] [Questions] [answered]
Is it efficient to create a new connection each time I need to access the database?

Or should I use a connection manager?

Which one should I use? some connection manager projects are not well maintained.

**Comment by gwenn** [comment, marked as answer]
> Is it efficient to create a new connection each time I need to access the database?
> Or should I use a connection manager?

It depends on many factors.

> Which one should I use? some connection manager projects are not well maintained.

[r2d2-sqlite](https://crates.io/crates/r2d2-sqlite) seems to be the most popular one.
